chatkit/src/main/java/com/pusher/chatkit/messages/MessageService.kt

```diff
@@ -91,17 +97,23 @@ class MessageService(
         tokenParams = tokenParams
     ).parseResponseWhenReady()
 
-    private fun sendCompleteMessage(
-        text: String? = null,
+    private fun sendMessage(
+        text: CharSequence = "",
         attachment: AttachmentBody
     ): MessageIdPromiseResult =
-        MessageRequest(text = text, userId = currentUser.id, attachment = attachment.takeIf { it !== AttachmentBody.None })
+        chatManager.currentUser.flatMapResult { currentUser ->
+            currentUser.sendMessage(text, attachment)
+        }
+
+    private fun CurrentUser.sendMessage(
+        text: CharSequence = "",
```

Comment:
Why a `CharSequence` instead of a `String`?

Reply:
`String` is a `CharSequence` Item 18: :D
http://jtechies.blogspot.co.uk/2012/07/item-18-prefer-interfaces-to-abstract.html